File: samples/src/main/java/com/compdfkit/samples/SampleListActivity.java
/**
 * Copyright © 2014-2025 PDF Technologies, Inc. All Rights Reserved.
 * <p>
 * THIS SOURCE CODE AND ANY ACCOMPANYING DOCUMENTATION ARE PROTECTED BY INTERNATIONAL COPYRIGHT LAW
 * AND MAY NOT BE RESOLD OR REDISTRIBUTED. USAGE IS BOUND TO THE ComPDFKit LICENSE AGREEMENT.
 * UNAUTHORIZED REPRODUCTION OR DISTRIBUTION IS SUBJECT TO CIVIL AND CRIMINAL PENALTIES.
 * This notice may not be removed from this file.
 */

package com.compdfkit.samples;

import android.os.Bundle;
import android.os.Environment;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.compdfkit.samples.util.FileUtils;

import java.io.File;

public class SampleListActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_sample_list);
        File file = getExternalFilesDir(Environment.DIRECTORY_DOCUMENTS);
        FileUtils.deleteFile(file);
        RecyclerView recyclerView = findViewById(R.id.rv_sample_list);

        SampleListAdapter listAdapter = new SampleListAdapter();
        recyclerView.setLayoutManager(new LinearLayoutManager(this));
        recyclerView.setAdapter(listAdapter);

        listAdapter.setOnItemClickListener(position -> {
            SampleDetailActivity.openDetail(this, position);
        });
    }
}

File: Samples_kotlin/src/main/java/com/compdfkit/samples/SampleListActivity.kt
/**
 * Copyright © 2014-2023 PDF Technologies, Inc. All Rights Reserved.
 *
 *
 * THIS SOURCE CODE AND ANY ACCOMPANYING DOCUMENTATION ARE PROTECTED BY INTERNATIONAL COPYRIGHT LAW
 * AND MAY NOT BE RESOLD OR REDISTRIBUTED. USAGE IS BOUND TO THE ComPDFKit LICENSE AGREEMENT.
 * UNAUTHORIZED REPRODUCTION OR DISTRIBUTION IS SUBJECT TO CIVIL AND CRIMINAL PENALTIES.
 * This notice may not be removed from this file.
 */
package com.compdfkit.samples

import android.os.Bundle
import android.os.Environment
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.compdfkit.core.document.CPDFAbility
import com.compdfkit.samples.util.FileUtils

class SampleListActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_sample_list)
        CPDFAbility.checkLicenseAllAbility()
        val file = getExternalFilesDir(Environment.DIRECTORY_DOCUMENTS)
        file?.let {
            FileUtils.deleteFile(it)
        }
        supportActionBar?.setTitle(R.string.samples)
        val recyclerView = findViewById<RecyclerView>(R.id.rv_sample_list)
        val listAdapter = SampleListAdapter().apply {
            onItemClickListener = { position ->
                SampleDetailActivity.openDetail(this@SampleListActivity, position)
            }
        }
        recyclerView.layoutManager = LinearLayoutManager(this)
        recyclerView.adapter = listAdapter
    }
}